net/dnscache/dnscache.go

```diff
@@ -119,14 +119,21 @@ func (r *Resolver) cloudHostResolver() (v *net.Resolver, ok bool) {
 		// which supports net.Resolver.PreferGo on Windows.
 		return nil, false
 	}
-	if cloudenv.Get() != cloudenv.GCP {
+	var ip string
+	switch cloudenv.Get() {
+	case cloudenv.GCP:
+		ip = cloudenv.GoogleMetadataAndDNSIP
```

Comment:
Use cloudenv.Get().ResolverIP() ? Saves you a bunch of switching and stuff.

Reply:
Right. That was a late addition. Will do.